libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -880,19 +880,43 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 			solAssert(_memberAccess.annotation().type, "_memberAccess has no type");
 			if (auto funType = dynamic_cast<FunctionType const*>(_memberAccess.annotation().type.get()))
 			{
-				if (funType->location() != FunctionType::Location::Internal)
-				{
-					_memberAccess.expression().accept(*this);
-					m_context << funType->externalIdentifier();
-				}
-				else
+				switch (funType->location())
 				{
+				case FunctionType::Location::Internal:
 					// We do not visit the expression here on purpose, because in the case of an
 					// internal library function call, this would push the library address forcing
 					// us to link against it although we actually do not need it.
-					auto const* function = dynamic_cast<FunctionDefinition const*>(_memberAccess.annotation().referencedDeclaration);
-					solAssert(!!function, "Function not found in member access");
-					utils().pushCombinedFunctionEntryLabel(*function);
+					if (auto const* function = dynamic_cast<FunctionDefinition const*>(_memberAccess.annotation().referencedDeclaration))
+						utils().pushCombinedFunctionEntryLabel(*function);
+					else
+						solAssert(false, "function not found");
```

Comment:
Probably we can keep the original assert text: `Function not found in member access`